Neuro/Impostor/ImpostorHandler.cs

```diff
@@ -0,0 +1,243 @@
+            {
+                goingForKill = true;
+                PlayerControl.LocalPlayer.StartCoroutine(MurderTarget());
+            }
+        }
+    }
+
+    [HideFromIl2Cpp]
+    private IEnumerator MurderTarget()
+    {
+        // wait a random amount of time so we dont just instantly kill them the second we are in range
+        yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.2f, 0.6f));
+        // this actually works in practice area which is pretty nice
+        PlayerControl.LocalPlayer.MurderPlayer(killTarget);
+        Info(String.Format("I just killed {0}!", killTarget.Data.PlayerName));
+        goingForKill = false;
+        killTarget = null;
+        NeuroPlugin.Instance.Tasks.UpdatePathToTask(NeuroPlugin.Instance.Tasks.GetFurthestTask());
+    }
+
+
+    [HideFromIl2Cpp]
+    private void AttemptVent()
+    {
+        // currently will just enter a vent whenever possible
+        // this should be changed to be more situational
+        if (ClosestVent != null && !PlayerControl.LocalPlayer.inVent && !PlayerControl.LocalPlayer.walkingToVent)
+        {
+            if (HudManager.Instance.ImpostorVentButton.currentTarget == ClosestVent)
+            {
+                // vent.EnterVent() and vent.Use() dont actually put you in the vent for whatever reason so just click the button virtually
+                HudManager.Instance.ImpostorVentButton.DoClick();
+            }
+        }
+    }
+
+    // example of vent functionality
+    public IEnumerator Vent(Vent original)
+    {
+        Info("I entered a vent!");
+        List<Vent> possibleVents = GetAvailableNearbyVents(original);
+        Vent current = possibleVents[UnityEngine.Random.RandomRangeInt(0, possibleVents.Count)];
+        yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.8f, 1.2f));
+        string error;
+        if (!original.TryMoveToVent(current, out error))
+        {
+            Error(String.Format("Failed to move to vent {0}, reason: {1}", current.Id, error));
+        }
+        while (true)
+        {
+            // use a random time between vent moves to make it more realistic
+            yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.8f, 1.2f));
+            bool playerFound = false;
+            foreach (KeyValuePair<PlayerControl, LastSeenPlayer> player in NeuroPlugin.Instance.Vision.GetPlayerLocations())
+            {
+                if (PlayerControl.LocalPlayer == player.Key) continue;
+                if (player.Key.Data.IsDead) continue;
+                if (player.Key.Data.RoleType == AmongUs.GameOptions.RoleTypes.Impostor) continue;
+
+                // if we see a player in our radius, try a different vent
+                if (player.Value.time > Time.timeSinceLevelLoad - (2 * Time.fixedDeltaTime))
+                {
+                    Info(String.Format("Spotted {0}, trying a different exit vent...", player.Key.name));
```

Comment:
I'm going to nitpick, we have interpolated strings everywhere else in the plugin.

Reply:
This code is part of a stub, so I'll probably just delete that line or leave it as is